capsule.sh

```diff
@@ -0,0 +1,267 @@
+#!/bin/bash
+
+set -e
+
+function die() {
+    echo error: "$@" 1>&2
+    exit 1
+}
+
+function confirm() {
+    local msg=$1
+    read -r -n1 -p "$msg [y/N] "
+    echo
+    if [ "$REPLY" != y ]; then
+        die "operation cancelled by the user"
+    fi
+}
+
+function print_banner() {
+    local msg=$1
+
+    echo
+    echo ========== "${msg^^*}" ==========
+    echo
+}
+
+function print_usage() {
+    echo "Usage: $(basename "$0") subcommand [subcommand-args...]"
+    echo
+    echo 'Subcommands:'
+    echo '  help    print this message'
+    echo '  keygen  use OpenSSL to auto-generate test keys suitable for signing'
+    echo '          positional argument: directory-path'
+    echo '  make    build a capsule, options:'
+    echo '          -t root-certificate-file'
+    echo '          -o subroot-certificate-file'
+    echo '          -s signing-certificate-file'
+}
+
+function help_subcommand() {
+    print_usage
+}
+
+function keygen_subcommand() {
+    local dir=$1
+
+    if [ $# -ne 1 ]; then
+        echo "Usage: $(basename "$0") keygen keys-dir"
+        exit 1
+    fi
+
+    if [ -e "$dir" ]; then
+        confirm "OK to remove '$dir'?"
+        rm -r "$dir"
+    fi
+
+    mkdir "$dir"
+    cd "$dir"
+
+    # this is needed to make `openssl req` work non-interactively
+    cat > "openssl.cnf" << EOF
+.include /etc/ssl/openssl.cnf
+
+[ req_root ]
+prompt             = no
+distinguished_name = req_root_dn
+x509_extensions    = v3_ca
+string_mask        = utf8only
+[ req_root_dn ]
+countryName         = XX
+stateOrProvinceName = Province
+organizationName    = Org
+commonName          = root
+
+[ req_sub ]
+prompt             = no
+distinguished_name = req_sub_dn
+x509_extensions    = v3_ca
+string_mask        = utf8only
+[ req_sub_dn ]
+countryName         = XX
+stateOrProvinceName = Province
+organizationName    = Org
+commonName          = sub
+
+[ req_sign ]
+prompt             = no
+distinguished_name = req_sign_dn
+x509_extensions    = v3_ca
+string_mask        = utf8only
+[ req_sign_dn ]
+countryName         = XX
+stateOrProvinceName = Province
+organizationName    = Org
+commonName          = sign
+EOF
+
+    print_banner 'Making root certificate'
+
+    # make root certificate
+    openssl genrsa -out root.p8e 2048
+    openssl req -config openssl.cnf -section req_root -new -x509 -days 3650 -key root.p8e -out root.pub.pem
```

Comment:
`-section` requires relatively new OpenSSL. Both `2023-11-24_2731fa619b` and `2024-02-18_732134932b` versions of coreboot-sdk used by `build.sh` on `dasharo` and `dasharo-4.21` respectively support it, but `2021-09-23_b0d87f753c` used by [CI on `dasharo-4.21` branch](https://github.com/Dasharo/coreboot/blob/dasharo-4.21/.github/workflows/build.yml#L17) does not.

As `dasharo` works in all cases, there is nothing to do here, but if some vendor wanted to have capsule updates on older branch, at least CI would have to be updated (which may generate another problems). We can always use it as an incentive to persuade them to switch to main branch (@macpijan @BeataZdunczyk CC), that would be less maintenance problems for us.

Reply:
The `-section` was needed only to create `openssl.cnf` once.  I can make 3 files (or reuse it 3 times) all with `req` section which should with any OpenSSL version.